src/Microsoft.PowerShell.Commands.Utility/commands/utility/EnableDisableRunspaceDebugCommand.cs

```diff
@@ -362,58 +378,63 @@ protected override void ProcessRecord()
             if (this.ParameterSetName.Equals(CommonRunspaceCommandBase.ProcessNameParameterSet))
             {
                 SetDebugPreferenceHelper(ProcessName, AppDomainName, true, "EnableRunspaceDebugCommandPersistDebugPreferenceFailure");
+                return;
             }
-            else
-            {
-                IReadOnlyList<Runspace> results = GetRunspaces();
 
-                foreach (var runspace in results)
+            IReadOnlyList<Runspace> results = GetRunspaces();
+
+            foreach (var runspace in results)
+            {
+                if (runspace.RunspaceStateInfo.State != RunspaceState.Opened)
                 {
-                    if (runspace.RunspaceStateInfo.State != RunspaceState.Opened)
-                    {
-                        WriteError(
-                            new ErrorRecord(new PSInvalidOperationException(string.Format(CultureInfo.InvariantCulture, Debugger.RunspaceOptionInvalidRunspaceState, runspace.Name)),
-                            "SetRunspaceDebugOptionCommandInvalidRunspaceState",
-                            ErrorCategory.InvalidOperation,
-                            this)
-                            );
+                    WriteError(
+                        new ErrorRecord(new PSInvalidOperationException(string.Format(CultureInfo.InvariantCulture, Debugger.RunspaceOptionInvalidRunspaceState, runspace.Name)),
+                        "SetRunspaceDebugOptionCommandInvalidRunspaceState",
+                        ErrorCategory.InvalidOperation,
+                        this)
+                        );
```

Comment:
Please move to end of line 394 (CodeFactor issue)

Reply:
fixed